src/dso/cli/_mv.py

```
@@ -0,0 +1,175 @@
+"""Rename stage or folder"""
+
+import os
+import sys
+from os import getcwd
+from pathlib import Path
+
+import rich_click as click
+from rich.prompt import Prompt
+
+from dso._logging import log
+from dso._util import get_project_root
+
+
+def update_all_references_to_source(project_root: Path, source_absolute_path: str, target_absolute_path: str):
+    """In the whole project, search for relative references to the old location and update to relative reference to the new location."""
+    log.debug("[yellow]Updating all references to source")
+    log.debug(f"{project_root} - {source_absolute_path} - {target_absolute_path}")
+    for dvc_dir in project_root.rglob("*"):
+        if dvc_dir.is_dir() and (dvc_dir / "dvc.yaml").exists() and dvc_dir != target_absolute_path:
+            source_direct_path = os.path.relpath(source_absolute_path, start=dvc_dir)
+            target_direct_path = os.path.relpath(target_absolute_path, start=dvc_dir)
+
+            log.info(f"Relatives path from {dvc_dir}: {source_direct_path} to {target_direct_path}")
+
+            update_references_in_file(dvc_dir / "dvc.yaml", source_direct_path, target_direct_path)
+            update_references_in_file(dvc_dir / "params.in.yaml", source_direct_path, target_direct_path)
+
+            # Iterate over all non-hidden files in the src subdirectory of dvc_dir
+            src_subdir = dvc_dir / "src"
+            if src_subdir.exists() and src_subdir.is_dir():
+                for file_path in src_subdir.rglob("*"):
+                    if file_path.is_file() and not file_path.name.startswith("."):
+                        update_references_in_file(file_path, source_direct_path, target_direct_path)
+
+
+def update_references_in_file(file: Path, pattern: str, replacement: str):
+    """Update all pattern in file"""
+    if os.path.exists(file):
```

Is it intended to silently do nothing if the file does not exist?